src/optimizer/query_graph.c

```diff
@@ -6177,11 +6156,23 @@ qo_classify_outerjoin_terms (QO_ENV * env)
 	}
 
       nidx_self = -1;		/* init */
+      is_outerjoin = false;
       for (t = bitset_iterate (&(QO_TERM_NODES (term)), &iter); t != -1; t = bitset_next_member (&iter))
 	{
 	  node = QO_ENV_NODE (env, t);
-
 	  nidx_self = MAX (nidx_self, QO_NODE_IDX (node));
+	  if (QO_TERM_IS_FLAGED (term, QO_TERM_OR_PRED) && QO_NODE_IS_OUTER_JOIN (node))
+	    {
+	      is_outerjoin = true;	/* for OR predicate */
+	    }
+	}
+      if (nidx_self != -1)
+	{
+	  node = QO_ENV_NODE (env, nidx_self);
+	  if (QO_NODE_IS_OUTER_JOIN (node))
+	    {
+	      is_outerjoin = true;
+	    }
```

Comment:
If "is_outerjoin" is already true, to check the condition "nidx_self != -1" looks useless because the "is_outerjoin" is not changed any more after this line.

Reply:
'is_outerjoin = false;' is missing. Thanks.